### ISSUE TITLE
native build: assembler fail to build objects due to fatal errors causing by unability to create /tmp/xxxxxxx.o files.

### ISSUE THREAD

**Issue by kas1e** [AmigaLabs/binutils-gdb#18]
When we tried to build anything via new native AS (by using gcc or directly), we do have:

```
Assembler messages:
Fatal error: can't create /tmp/xxxxxxx.o : No such file or directory
```

Probably that the "unix paths" issue, and we should add -lunix at the end of compiling lines, but not sure if it all what need to be done to be deal with.

**Comment by kas1e** [comment]
Ok this one fixed by adding -lunix to LDFLAGS, can be closed.

**Comment by kas1e** [comment]
This one again there, as in the latest commits, ```LDFLAGS="-lunix"``` were deleted by accident from the native build instructions from the main README.

**Comment by migthymax** [comment]
Because gdb now uses clib4, it is even enough to put a `.unix`file into the bin directory

**Comment by kas1e** [comment]
Yeah, but IMHO in long run we need to fix it "inside" (imho) ? As all in all this is the native build, so we had to care for all the pathes/slashes/progdirs/homes/etc ? Not that it's big deal for now, but probably for some "public kind of final" release should be done (imho again :) )

**Comment by migthymax** [comment]
Yes you right, but after trying to deal with this issue and thinking about it. The issue is that the as is called with `/tmp/...`files. You can see this with running `gcc -v` in verbose mode. And os I think the issue is with the gcc compiler resp. the collect command. But nevertheless it should work as a whole package in the end.

**Comment by kas1e** [comment]
Tested more , and i can't deal with it if i just do `assign tmp: ram:' , same error. I also can't deal with it by putting empty .unix file in the directory from which i tried to build this example.  And even if i tried to test this issue by compiling sources from within "sh" shell, it still the same.

**Comment by migthymax** [comment]
I stumbled into this too. The `.unix` file must be put into the directory where the `ld`,`ar`... binaries are located.
In my case something like this: `GCC:ppc-amigaos/bin/`

**Comment by walkero-gr** [comment]
By adding the following in the `main()` in `gas/as.c`, this seems to be solved
```
#if defined(__amigaos4__) && defined(__CLIB4__)
  enableUnixPaths();
#endif
 ```
but then the `ld` has the following problems

![Image](https://github.com/user-attachments/assets/8707c56d-253f-43df-b48d-cbb0093a8085)

**Comment by migthymax** [comment]
I think the issue with the ld comes from https://github.com/AmigaLabs/binutils-gdb/issues/34

**Comment by migthymax** [comment]
All Amiga specific file/path handling has now been disabled with an `#ifdef ENABLE_HAVE_AMIGA_BASED_FILE_SYSTEM`
So if one want to have tools which can handle Amiga paths the following define has to be past in during compilation`-DENABLE_HAVE_AMIGA_BASED_FILE_SYSTEM`. BTW the Amiga  specific path handling is still buggy.
Addition all tools like ar,as,etc now embed as first call in main:
```
#if defined(__amigaos4__) && defined(__CLIB4__)
  enableUnixPaths();
#endif
```
So that the user don't need to put a `.unix`file anymore in the program directory.

**Comment by walkero-gr** [comment]
@migthymax Thank you for this update.
Based on a discussion we had on AmigaLabs discord, we proposed that for this kind of tools, we should avoid using the `enableUnixPaths()` and do a proper support of Amiga paths. This means, if I get it right, that we need to use the AmigaOS API to create and access files.
How much of work do you think this could be?
Also, I believe we do not need to have a new definition, like the `ENABLE_HAVE_AMIGA_BASED_FILE_SYSTEM` since we can check against `__amigaos4__`. What do you think about it?

**Comment by migthymax** [comment]
@walkero-gr 
I missed the discussion, but I see it in the same way.

The current solution to disable the amiga based file system, is only temp. and to avoid the lost of the code. The now disabled implementation still has issues (or better not yet tested) with the current directory which is `""` on amiga and handling of parent up walks like `//`.

That said, having binutils  amiga path aware is just the have the work. The other work is to transform the complete toolchain. This issue actually arise because the `gcc`resp. `collect` aren't aware of the amiga paths and calls the tools with unix path like `/tmp/jkkjk.o` instead of `TMP:jkkjk.o`

**Comment by josefwegner** [comment]
Seems to work with nightly:
```
    .section ".text"
    .globl _start

_start:
    li      3, 0      # Rückgabewert 0
    blr                 # zurück zum Loader -> Programmende
```

```
as -mregnames -o test.o test.s
ld -o test test.o
```
